### DISCUSSION TITLE
Newport City configuration not working

### DISCUSSION THREAD

**Discussion by andyevans-eng** [mampfes/hacs_waste_collection_schedule#2271] [Q&A]
Everytime I try to configure it to my local source using the GUI I get the following error: The source returned an invalid response: "500 Server Error: Internal Server Error for url: https://api.newport.gov.uk/PropertyData/bincollection/properties/1000*******/Waste%20Collection". Please check the provided arguments and try again.
If I copy that URL into the browser I get http error 500 this page is not working.
(I have hidden my UPRN)

**Comment by DeadBadger** [comment]
I've tried setting this up today and, coincidently, live in Newport too. I think it might be because Newport city have been updating their website and the current settings are out of date

**Comment by 5ila5** [comment]
Do you still encounter this problem? The source seems to work just fine on my end with uprn=100100688837

**Comment by andyevans-eng** [comment]
I tried again this morning and get this error:  The source returned an invalid response: "500 Server Error: Internal Server Error for url: https://api.newport.gov.uk/PropertyData/bincollection/properties/10009645040/Waste%20Collection". Please check the provided arguments and try again.

 

I managed to get it working with the downloadable ICS file, but if the council change the schedule I will have to update the ICS file. Would be much better if the GUI setup worked as then it would always be up to date.

 

I am in NP10 area, I also had a message from someone who lives near me and it worked perfectly for them.

 

Regards,

Andy

**Reply by DeadBadger** [reply]
I've just set up the calendar with the ICS file which is fine for my home display but would much prefer to use the Waste Collection GUI as it looks nice and clean for my mobile Home assistant dashboard. Fingers crossed they get it working. I'm also by Asda Duffryn and am turning into a home assistant/smart home junkie lol

**Reply by 5ila5** [reply]
You can use the ICS file in the GUI configuration coutnry: gerneric -> ics set file

**Reply by DeadBadger** [reply]
The problem I'm having, to be fair I'm a complete idiot, is after selecting ICS File as an option, it wants either the URL or the name/location of the ICS file and whatever I put in is failing

**Comment by andyevans-eng** [comment]
Hi,

 

I just tried again and still had the same error:    The source returned an invalid response: "500 Server Error: Internal Server Error for url: https://api.newport.gov.uk/PropertyData/bincollection/properties/10009645040/Waste%20Collection". Please check the provided arguments and try again.

 

When did you set this up? I am wondering if council website has been updated since you set yours up?

I see you are on Stow Hill, I am by Asda Duffryn.

I am loving this Home Assistant and always finding new things it can do.

 

Regards,

 

Andy

**Comment by 5ila5** [comment]
Looks like the main page moved over to https://iportal.itouchvision.com our test case seem to still work but maybe not all address.

We have a source for https://iweb.itouchvision.com but it looks like they don't do the same requests. I tried to figure out what https://iportal.itouchvision.com does but they are sending weird binary data I cannot make any sense out of. The JavaScript dealing with these binary data is 50,000 lines of code probably somehow compiled out of a framwork as the variable/function names are just 2 letters or numbers. Looks like it's a bigger project trying to reverse engineer what they are doing there. You can meanwhile use the ics source with the downloaded ICS file from the website (Add to your calendar link after searching your address)

**Comment by 5ila5** [comment]
@ReneNulschDE added it to the itouchvision source and removed the dedicated Newport source in #2293, You will be able to use the [itouchvision](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/source/iweb_itouchvision_com.md) source after the next release (or now by changing to the master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)

**Comment by andyevans-eng** [comment]
The way I set it up was by using the file editor and uploading the ics file into the www folder, then where it asks for the URL enter: http://your_HA_IP/local/collections.ics

 

(collections.ics being the name of my file) Also, you may need to use https instead of http, depending on how you have it set up, just use the address you use to access HA with /local/yourfilename.ics at the end

**Reply by DeadBadger** [reply]
That has worked for me, thank you so much
